File: api/proxy.py
from fastapi import FastAPI, Response
import httpx

app = FastAPI()

@app.get("/")
async def proxy():
    async with httpx.AsyncClient() as client:
        resp = await client.get("https://studyuk.fun/goal/")
        html = resp.text

    # Replace Telegram links
    html = html.replace(
        "https://t.me/",
        "https://t.me/+rc5Psv_S2VJkMGM1"
    )

    return Response(content=html, media_type="text/html")

File: api/proxy.js
export default async function handler(req, res) {
  try {
    const response = await fetch('https://studyuk.fun/goal/', {
      headers: { 'User-Agent': req.headers['user-agent'] || '' }
    });
    const buffer = await response.arrayBuffer();
    const html = new TextDecoder('utf-8').decode(buffer);

    // Remove all <script> tags to avoid atob error
    let clean = html.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '');

    // Replace Telegram links with your custom one
    clean = clean.replace(/https:\/\/t\.me\/[^\s"'<>]+/g, 'https://t.me/+rc5Psv_S2VJkMGM1');

    // ✅ Correct hyphen here
    res.setHeader('Content-Type', 'text/html; charset=utf-8');
    res.setHeader('Cache-Control', 's-maxage=60, stale-while-revalidate');

    res.status(200).end(clean);
  } catch (e) {
    res.status(500).end('Proxy error: ' + e.message);
  }
}
